src/v/redpanda/admin_server.cc

```diff
@@ -912,14 +912,24 @@ ss::future<> admin_server::throw_on_error(
               fmt_with_ctx(fmt::format, "Can not find pid for ntp:{}", ntp));
         case cluster::tx_errc::partition_not_found: {
             ss::sstring error_msg;
-            if (ntp == model::tx_manager_ntp) {
+            if (
+              ntp.tp.topic == model::tx_manager_topic
+              && ntp.ns == model::kafka_internal_namespace) {
                 error_msg = fmt::format("Can not find ntp:{}", ntp);
             } else {
                 error_msg = fmt::format(
                   "Can not find partition({}) in transaction for delete", ntp);
             }
             throw ss::httpd::bad_request_exception(error_msg);
         }
+        case cluster::tx_errc::not_coordinator:
+            throw ss::httpd::base_exception(
+              fmt::format(
+                "Node not a coordinator or coordinator leader is not "
+                "stabilized yet: {}",
+                ec.message()),
+              ss::httpd::reply::status_type::service_unavailable);
```

Comment:
this going to break in upstream/dev due to changes in seastar that merged earlier today. needs to be in the `http` namespace rather than `httpd`.